workflows/train_bandit.py

```diff
@@ -76,6 +97,34 @@ def get_feature_order(features_spec: Dict) -> List[str]:
     return float_feature_order, id_feature_order
 
 
+def preprocess_feature(
+    feature_name: str, feature_type: str, values: np.array
+) -> Tuple[pd.DataFrame, int]:
+    assert feature_type in ("N", "C")
+
+    # convert to scikit learn expected format
+    values = values.reshape(-1, 1)
+
+    if feature_type == "N":
+        # standard scaler seems to be right choice for numeric features
+        # in regression problems
+        # http://rajeshmahajan.com/standard-scaler-v-min-max-scaler-machine-learning/
+        preprocessor = preprocessing.StandardScaler()
+        values = preprocessor.fit_transform(values)
+        df = pd.DataFrame(values.squeeze(), columns=[feature_name])
+    elif feature_type == "C":
+        preprocessor = preprocessing.OneHotEncoder(sparse=False)
```

Comment:
so all of this logic is for features on products, yeah? am i understanding this correctly - we 1-hot a categorical feature like ["large", "medium", "small"] and use standardscaler on numerical (e.g. plant height)?

Reply:
yup!